ffs/filcold/filcold.go

```diff
@@ -269,9 +269,12 @@ func (fc *FilCold) makeDeals(ctx context.Context, c cid.Cid, size uint64, cfgs [
 // If the deal finishes successfully it returns a FilStorage result.
 // If the deal finished with error, it returns a ffs.DealError error
 // result, so it should be considered in error handling.
-func (fc *FilCold) WaitForDeal(ctx context.Context, c cid.Cid, proposal cid.Cid, timeout time.Duration) (ffs.FilStorage, error) {
+func (fc *FilCold) WaitForDeal(ctx context.Context, c cid.Cid, proposal cid.Cid, timeout time.Duration, dealUpdates chan<- ffs.FilStorage) (ffs.FilStorage, error) {
```

Comment:
The `chan<- ffs.FilStorage` is passed all the way down into this low level func that waits for a deal to become final.

Reply:
Might worth adding in the comments that this method will close `dealUpdates` channel that is received as a param, so the caller shouldn't since that would panic.